openstack/networking/v2/extensions/layer3/floatingips/results.go

```diff
@@ -136,3 +136,73 @@ func ExtractFloatingIPs(r pagination.Page) ([]FloatingIP, error) {
 func ExtractFloatingIPsInto(r pagination.Page, v interface{}) error {
 	return r.(FloatingIPPage).Result.ExtractIntoSlicePtr(v, "floatingips")
 }
+
+type PortForwarding struct {
+	// The ID of the floating IP port forwarding
+	ID string `json:"id"`
+
+	// The ID of the Neutron port associated to the floating IP port forwarding.
+	InternalPortID string `json:"internal_port_id"`
+
+	// The TCP/UDP/other protocol port number of the port forwarding’s floating IP address.
+	ExternalPort int `json:"external_port"`
+
+	// The IP protocol used in the floating IP port forwarding.
+	Protocol string `json:"protocol"`
+
+	// The TCP/UDP/other protocol port number of the Neutron port fixed
+	// IP address associated to the floating ip port forwarding.
+	InternalPort int `json:"internal_port"`
+
+	// The fixed IPv4 address of the Neutron port associated
+	// to the floating IP port forwarding.
+	InternalIPAddress string `json:"internal_ip_address"`
+}
+
+// ExtractPortForwarding will extract a Port Forwarding resource from a result.
+func (r commonResult) ExtractPortForwarding() (*PortForwarding, error) {
```

Comment:
This is inheriting from the Floating IP create result which is different than the PortForwarding result.

To prevent overlap like this, I recommend creating a new neutron extension at `openstack/networking/v2/extensions/layer3/portforwarding`.